applications/StructuralMechanicsApplication/tests/test_eigen_solver_with_mass_normalized_eigenmodes.py

```diff
@@ -0,0 +1,136 @@
+import KratosMultiphysics
+
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+from KratosMultiphysics.StructuralMechanicsApplication.structural_mechanics_analysis import StructuralMechanicsAnalysis
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+# External imports
+import os
+import json
+
+'''
+Test description:
+This test does an eigenvalue analysis on a simple cantilever beam, and compares the mass
+normalized eigenmodes with a reference solution
+'''
+
+def GetFilePath(file_name):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), file_name)
+
+@KratosUnittest.skipIfApplicationsNotAvailable("LinearSolversApplication")
+class TestEigenSolverWithMassNormalizedEigenmodes(KratosUnittest.TestCase):
+    # muting the output
+    KratosMultiphysics.Logger.GetDefaultOutput().SetSeverity(KratosMultiphysics.Logger.Severity.WARNING)
+
+    def test_eigen_with_mass_normalized_eigenmodes(self):
+        self.execute_test_eigen_with_mass_normalized_eigenmodes(use_block_builder=False)
+
+    def execute_test_eigen_with_mass_normalized_eigenmodes(self, use_block_builder):
+        analysis_parameters = KratosMultiphysics.Parameters("""{
+            "problem_data"    : {
+                "parallel_type" : "OpenMP",
+                "echo_level"    : 1,
+                "start_time"    : 0.0,
+                "end_time"      : 1.0
+            },
+            "solver_settings" : {
+                "solver_type"              : "eigen_value",
+                "model_part_name"          : "Structure",
+                "domain_size"              : 3,
+                "model_import_settings"    : {
+                    "input_type"     : "use_input_model_part"
+                },
+                "time_stepping"            : {
+                    "time_step" : 1.1
+                },
+                "normalize_eigenvectors_with_mass_matrix": true,
+                "rotation_dofs"            : true,
+                "block_builder"            : true
+            }
+        }""")
+
+        analysis_parameters["solver_settings"]["block_builder"].SetBool(use_block_builder)
+        
+        # Create the model, the analysis and run the eigenvalue problem
+        model = KratosMultiphysics.Model()
+        analysis = StructuralMechanicsAnalysis(model, analysis_parameters.Clone())
+        model_part = model["Structure"]
+        SetupSystem(model_part)
+        analysis.Run()
+
+        # Get the reference file name
+        reference_file = self.__GetReferenceFileName()
```

Comment:
I don't want to be super picky, but I would have done this using the following to navigate to the testing folder.

https://github.com/KratosMultiphysics/Kratos/blob/master/applications%2FOptimizationApplication%2Ftests%2Falgorithm_tests%2Fanalysis_based_tests%2Falgorithm_steepest_descent%2Ftest_steepest_descent_analysis.py#L10